### PR TITLE
Add more outputs to fetch node including status code and computing json schema

### PR DESCRIPTION
**Pull request by Mearman**
image

### CODE REVIEW

File: packages/core-kit/src/nodes/fetch.ts [hunk 1]
```diff
@@ -118,12 +119,54 @@ export const fetchDescriber: NodeDescriberFunction = async () => {
...
+        },
       },
-      required: ["response", "status"],
+      required: ["response"],
     },
   };
 };
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function generateJsonSchemaFromObject(obj: any): Schema {
```
Comment (dglazkov) [inline packages/core-kit/src/nodes/fetch.ts:152]: use unknown here

File: packages/core-kit/src/nodes/fetch.ts [hunk 2]
```diff
@@ -163,8 +217,28 @@ export default {
...
+        return {
+          response,
+          status,
+          statusText,
+          contentType,
+          responseHeaders,
+        };
+      } else {
+        response = await data.json();
+        const responseSchema = generateJsonSchemaFromObject(response);
```
Comment (dglazkov) [inline packages/core-kit/src/nodes/fetch.ts:232]: not sure about this one

**Review by dglazkov — CHANGES_REQUESTED** [review]

File: packages/core-kit/src/nodes/fetch.ts (hunk 1 above)
Comment (Mearman) [reply packages/core-kit/src/nodes/fetch.ts:152]: had that originally. but then TS is unhappy with recurse down https://github.com/ExaDev/breadboard/blob/fetch-node-outputs/packages/core-kit/src/nodes/fetch.ts#L161

File: packages/core-kit/src/nodes/fetch.ts (hunk 2 above)
Comment (Mearman) [reply packages/core-kit/src/nodes/fetch.ts:232]: which bit?

Comment (dglazkov) [reply packages/core-kit/src/nodes/fetch.ts:232]: The generating JSON schema of the response bit. Not sure why we would need that.

File: packages/core-kit/src/nodes/fetch.ts (hunk 1 above)
Comment (dglazkov) [reply packages/core-kit/src/nodes/fetch.ts:152]: Maybe use `NodeValue` here?

File: packages/core-kit/src/nodes/fetch.ts (hunk 2 above)
Comment (Mearman) [reply packages/core-kit/src/nodes/fetch.ts:232]: So two use-cases I have right now
1. To feed into the object and array manipulation nodes for picking apart a response
2. For checking and then inputting the schema when dealing with a new response

File: packages/core-kit/src/nodes/fetch.ts (hunk 1 above)
Comment (Mearman) [reply packages/core-kit/src/nodes/fetch.ts:152]: Tried that one as well
```bash
  No index signature with a parameter of type 'string' was found on type 'Capability | { [key: string]: NodeValue; }'.ts(7053)
```

File: packages/core-kit/src/nodes/fetch.ts (hunk 2 above)
Comment (dglazkov) [reply packages/core-kit/src/nodes/fetch.ts:232]: Got it. My spidey sense tells me that this is probably a different kind of node. Maybe a util node that takes a JSON object and produces a schema for it? Let's not add this bit to the fetch node.

File: packages/core-kit/src/nodes/fetch.ts (hunk 1 above)
Comment (dglazkov) [reply packages/core-kit/src/nodes/fetch.ts:152]: > had that originally. but then TS is unhappy with recurse down https://github.com/ExaDev/breadboard/blob/fetch-node-outputs/packages/core-kit/src/nodes/fetch.ts#L161

What error does it give? I would be surprised if it were unhappy with the conversion to "unknown"

File: packages/core-kit/src/nodes/fetch.ts (hunk 2 above)
Comment (Mearman) [reply packages/core-kit/src/nodes/fetch.ts:232]: brilliant idea

**Pull request merged**